complex/ft_config.c

```diff
@@ -40,8 +41,38 @@
 #define FT_MODE_ALL	/*FI_CONTEXT |*/ FI_LOCAL_MR /*| FI_MSG_PREFIX*/
 #define FT_MODE_NONE	~0ULL
 
+#define STRINGIFY(x) #x
+#define STREQ(STR1, STR2) !strncmp(STR1, STR2, strlen(STR2))
+
+#define TEST_ENUM_SET_N_RETURN(str, enum_val, type, key)	\
+	do {							\
+		if (STREQ(str, STRINGIFY(enum_val))) {		\
+			*(type *)(key) = enum_val;		\
+			return 0;				\
+		}						\
+	} while (0)
+
+#define TEST_SET_N_RETURN(str, val_str, val, type, key)	\
+	do {						\
+		if (STREQ(str, val_str)) {		\
+			*(type *)(key) = val;		\
+			return 0;			\
+		}					\
+	} while (0)
```

Comment:
The above macros look like they should build on each other.  Please move under common.